rmw_implementation/rmw_implementation-extras.cmake.in

```diff
@@ -15,69 +15,57 @@
 # generated from rmw_implementation/rmw_implementation-extras.cmake.in
 
 find_package(rmw_implementation_cmake REQUIRED)
-set(default_rmw_implementation "@RMW_IMPLEMENTATION@")
-set(available_rmw_implementations "@RMW_IMPLEMENTATIONS@")
 
 if(NOT "${RMW_IMPLEMENTATION}" STREQUAL "")
-  set(requested_rmw_implementation ${RMW_IMPLEMENTATION})
-  if(@RMW_IMPLEMENTATION_SUPPORTS_POCO@)
-    if(NOT requested_rmw_implementation IN_LIST available_rmw_implementations)
-      message(FATAL_ERROR
-        "The RMW implementation has been specified as "
-        "'${requested_rmw_implementation}' through CMake, "
-        "however it is not in the list of supported rmw implementations, "
-        "which was specified when the 'rmw_implementation' package was built.")
-    endif()
-  else()
-    if(NOT requested_rmw_implementation STREQUAL default_rmw_implementation)
-      message(FATAL_ERROR
-        "The RMW implementation has been specified as "
-        "'${requested_rmw_implementation}' through CMake, "
-        "however this needs to match the RMW implementation "
-        "'${default_rmw_implementation}', "
-        "which was specified when the 'rmw_implementation' package was built.")
-    endif()
-  endif()
+  set(requested_rmw_implementation "${RMW_IMPLEMENTATION}")
+  set(requested_rmw_implementation_from "CMake")
+elseif(NOT "$ENV{RMW_IMPLEMENTATION}" STREQUAL "")
+  set(requested_rmw_implementation "$ENV{RMW_IMPLEMENTATION}")
+  set(requested_rmw_implementation_from
+    "environment variable 'RMW_IMPLEMENTATION'")
+else()
+  set(requested_rmw_implementation "@RMW_IMPLEMENTATION@")
+  set(requested_rmw_implementation_from
+    "the default when @PROJECT_NAME@ wasbuilt")
 endif()
 
-if(NOT "$ENV{RMW_IMPLEMENTATION}" STREQUAL "")
-  set(requested_rmw_implementation $ENV{RMW_IMPLEMENTATION})
-  if(@RMW_IMPLEMENTATION_SUPPORTS_POCO@)
-    if(NOT requested_rmw_implementation IN_LIST available_rmw_implementations)
-      message(FATAL_ERROR
-        "The RMW implementation has been specified as "
-        "'${requested_rmw_implementation}' through the environment variable "
-        "'RMW_IMPLEMENTATION', "
-        "however it is not in the list of supported rmw implementations, "
-        "which was specified when the 'rmw_implementation' package was built.")
-    endif()
-  else()
-    if(NOT requested_rmw_implementation STREQUAL default_rmw_implementation)
-      message(FATAL_ERROR
-        "The RMW implementation has been specified as "
-        "'${requested_rmw_implementation}' "
-        "through the environment variable 'RMW_IMPLEMENTATION', "
-        "however this needs to match the RMW implementation "
-        "'${default_rmw_implementation}', "
-        "which was specified when the 'rmw_implementation' package was built.")
-    endif()
+if(@RMW_IMPLEMENTATION_BAKED@)
+  message(STATUS "Using RMW implementation '@RMW_IMPLEMENTATION@'")
+  if(NOT requested_rmw_implementation STREQUAL @RMW_IMPLEMENTATION@)
+    message(ERROR
```

Comment:
Does it make sense to continue in this case? We can't satisfy the explicitly requested configuration. I would argue a `FATAL_ERROR` message would be more appropriate here. Btw. that was also the behavior before.